Release 3.0.2

denihs opened the pull request:
Current released version: 3.0.2
To use it, please do a

```
meteor update --release 3.0.2
```

you can also use the following command to install this meteor version
```
npx meteor
```

## Highlights

With `3.0.2-beta.0`: 
- [Remove pinned version for the package react-meteor-data from our skeletons](https://github.com/meteor/meteor/commit/2fb7ccc5cf310f740776384748bd406e9502302b)
- `less@4.1.1`: [Publishing the official version ](https://github.com/meteor/meteor/pull/13237/commits/881ec31bd000b1f71d54bd7916ff7b1411a10517)
- https://github.com/meteor/meteor/pull/13236
- https://github.com/meteor/meteor/pull/13247
- https://github.com/meteor/meteor/pull/13288

JorgenVatle committed:
Track registered Mongo collections.

The idea behind this is to ease with package development where collections cannot always be imported explicitly.

The ability to retrieve a collection by name should ease with some edge cases where package authors rely on shims for the Mongo.Collection global in order to add functionality.

cultofcoders:redis-oplog is a good example of where this doesn't always work out well. Collections registered by compiler/build plugin packages appear to be registered before standard feature packages. This leads to the oplog package having no real way to reliably access those collection instances.

StorytellerCZ committed:
Merge branch 'devel' into mongo-feature-suggestion

JorgenVatle committed:
Assign `Mongo.Collections` map as `Mongo._collections`

JorgenVatle committed:
Attach public collection get() method on Mongo.Collection

JorgenVatle committed:
Extend summary block for Mongo.Collection.get() method

JorgenVatle committed:
Add JSDoc properties for static `Mongo.Collection.get()` method.

JorgenVatle committed:
Update Collection.get() summary with references to related resources.

JorgenVatle committed:
Add comment to `Collection.get()` name parameter

JorgenVatle committed:
Include `Mongo.Collection.get` in Collections API docs

JorgenVatle committed:
Correct documentation link for `Collection#rawDatabase()`

JorgenVatle committed:
Add type definition for `Mongo.Collection.get()`

JorgenVatle committed:
Remove redundant function name from `Mongo.Collection.get`

JorgenVatle committed:
Improve placement of static `Collection.get()` method.

JorgenVatle committed:
Move _collections into static namespace

JorgenVatle committed:
Use `this` context when accessing collections map

JorgenVatle committed:
Update collection tracker to match new API structure.

JorgenVatle committed:
Assert that collections are retrievable by name

JorgenVatle committed:
Merge branch 'devel' into mongo-feature-suggestion

JorgenVatle committed:
Simplify `Mongo.Collection.get()` to `Mongo.collection()`

JorgenVatle committed:
Update old `Mongo.collection.get()` documentation reference

JorgenVatle committed:
Merge remote-tracking branch 'origin/mongo-feature-suggestion' into mongo-feature-suggestion

JorgenVatle committed:
Rename Mongo.collection() get method to `Mongo.getCollection()`

JorgenVatle committed:
Update type definition to match new `Mongo.getCollection()` method.

StorytellerCZ committed:
Merge branch 'devel' into mongo-feature-suggestion

leonardoventurini committed:
fix subscription stopping too early

denihs committed:
Remove pinned version for the package react-meteor-data from our skeletons

denihs committed:
- publish official version for `less` and 4.1.1

denihs committed:
- update versions for meteortesting:mocha-core and meteortesting:mocha

denihs committed:
- update versions for meteortesting:mocha

denihs committed:
- fix `regressions - web.browser.legacy` test

nachocodoner committed:
update docs

leonardoventurini committed:
run operations in parallel

leonardoventurini committed:
add test

leonardoventurini committed:
shrinkwraps

leonardoventurini committed:
clean code

leonardoventurini committed:
fix conflicting publication name

leonardoventurini committed:
remove test not needed

leonardoventurini committed:
Revert "remove test not needed"

This reverts commit 6061c36bf8ca181a7a19893acfeb574fd682f53a.

leonardoventurini committed:
unsub still needs to be queued

leonardoventurini committed:
Merge pull request #13236 from meteor/fix/docs-removed-subs

Fix subscription still resetting documents

leonardoventurini committed:
preemptively remove code

leonardoventurini committed:
add command

leonardoventurini committed:
add message

leonardoventurini committed:
commit shrinkwrap

leonardoventurini committed:
Revert "preemptively remove code"

This reverts commit 94f7b4a6a1c690d535b5daf411b77723c1ddd0af.

leonardoventurini committed:
remove ready call

leonardoventurini committed:
Revert "remove ready call"

This reverts commit 0934440e090980573b142e921c503713a3cc190e.

leonardoventurini committed:
update auth in subs in a controlled manner

StorytellerCZ commented:
@denihs @nachocodoner 
https://github.com/nodejs/node/releases/tag/v20.16.0 just dropped

leonardoventurini committed:
fix missing option exception

leonardoventurini committed:
wait for user set id in methods

leonardoventurini committed:
only start auth sub once connected

leonardoventurini committed:
remove test

leonardoventurini committed:
make connection utility to capture messages

leonardoventurini committed:
add test

StorytellerCZ committed:
Merge branch 'devel' into release-3.0.2

leonardoventurini committed:
fix test due to conflict with other tests

leonardoventurini committed:
use built-in sleep

leonardoventurini committed:
use proper filter

leonardoventurini committed:
Merge pull request #13247 from meteor/fix/double-ready-auth-issue

Fix auth sub sending ready twice on load

leonardoventurini committed:
remove comments

denihs committed:
Meteor version to 3.0.2-beta.0 :comet:

StorytellerCZ commented:
Babel 7.25 is out as well: https://babeljs.io/blog/2024/07/26/7.25.0
Probably good idea to pair it together with the Node udpate.
https://github.com/babel/babel/releases/tag/v7.25.0

denihs committed:
Meteor version to 3.0.2-beta.1 :comet:

denihs committed:
Meteor version to 3.0.2-beta.2 :comet:

denihs committed:
publish version 3.0.2-beta.0 for the meteor installer package

leonardoventurini committed:
queue sending sub messages after outstanding methods

leonardoventurini committed:
revert autorun change

nachocodoner committed:
ensure to git reset checkout before updating git submodule

nachocodoner committed:
ensure to git reset checkout on all submodules as well

leonardoventurini committed:
adjust tests

nachocodoner committed:
Merge branch 'devel' into mongo-feature-suggestion

leonardoventurini committed:
Merge pull request #13174 from JorgenVatle/mongo-feature-suggestion

Feature suggestion: Track registered Mongo collection instances.

StorytellerCZ committed:
Remove version constraints from http package in accounts-twitter

Is it event needed anymore?

leonardoventurini committed:
Merge pull request #13262 from meteor/v3/login-sub-order

Fix outstanding subscription init order generally

denihs committed:
Merge pull request #13268 from meteor/feature/fix-http-in-twitter-oauth

Remove version constraints from http package in accounts-twitter

denihs committed:
Meteor version to 3.0.2-beta.3 :comet:

denihs committed:
remove http from accounts-twitter. Not used there.

denihs committed:
Meteor version to 3.0.2-beta.4 :comet:

denihs committed:
publish version 3.0.2-beta.4 for the meteor installer package

leonardoventurini committed:
Merge branch 'devel' into release-3.0.2

jordanbtucker committed:
Use PowerShell to set PATH instead of setx on Windows (#13254)

* fix(installer): use powershell to set PATH instead of setx

* style: run prettier

* style(installer): fix comma

leonardoventurini committed:
adjust readme and simplify wording and order

leonardoventurini committed:
adjust wording

leonardoventurini committed:
implement feedback

leonardoventurini committed:
remove global install command

nachocodoner committed:
Merge branch 'devel' into release-3.0.2

camstuart committed:
fix: typo

leonardoventurini committed:
Merge pull request #13280 from meteor/v3/update-installer-readme

Adjust Meteor Installer Readme

leonardoventurini committed:
Merge pull request #13285 from camstuart/docs-typo

V3 Docs landing page typo

OleksandrChekhovskyi committed:
Fix Meteor.promisify and add a test for it (#13284)

This fixes a regression introduced in commit 87f8697941.

Co-authored-by: Nacho Codoñer <igcogi@gmail.com>

nachocodoner committed:
Fix deprecation dependency on installer (#13289)

* fix deprecation dependency on installer

* re-run checks

jhernandez-etraid committed:
add ddpUrl to VALID_CONFIG_KEYS (#13296)

* add ddpUrl to VALID_CONFIG_KEYS

* 'ddpUrl'

---------

Co-authored-by: Nacho Codoñer <igcogi@gmail.com>

nachocodoner committed:
Deprecate Meteor.user() usage on server side (#13288)

nachocodoner committed:
Merge branch 'devel' into release-3.0.2

denihs committed:
Meteor version to 3.0.2 :comet:

denihs committed:
update npm-shrinkwrap.json files

nachocodoner committed:
re-run checks

nachocodoner committed:
Merge branch 'release-3.0.2' of https://github.com/meteor/meteor into release-3.0.2

denihs committed:
publish version 3.0.2 of the meteor installer

denihs committed:
Merge remote-tracking branch 'origin/release-3.0.2' into release-3.0.2

denihs committed:
update README.md

denihs committed:
publish a new version for email package

This was necessary because we had an issue publishing the email version when releasing 3.0.2. So we manually forced a new patch. Now apps can use this package again.
